src/config.py

```diff
@@ -135,3 +143,166 @@ def get_sagemaker_remote_efa_instance_type():
     Get the config value for sagemaker_remote_efa_instance_type
     """
     return parse_dlc_developer_configs("test", "sagemaker_remote_efa_instance_type")
+
+
+def is_pr_build_job_generic():
+    """
+    Return true if the build job is not dedicated to one flavor of image
+    :return: bool True/False
+    """
+    build_job_is_ei_dedicated = os.getenv("EIA_DEDICATED", "false").lower() == "true"
+    build_job_is_neuron_dedicated = os.getenv("NEURON_DEDICATED", "false").lower() == "true"
+    build_job_is_neuronx_dedicated = os.getenv("NEURONX_DEDICATED", "false").lower() == "true"
+    build_job_is_graviton_dedicated = os.getenv("GRAVITON_DEDICATED", "false").lower() == "true"
+    build_job_is_habana_dedicated = os.getenv("HABANA_DEDICATED", "false").lower() == "true"
+    build_job_is_hf_trcomp_dedicated = (
+        os.getenv("HUGGINFACE_TRCOMP_DEDICATED", "false").lower() == "true"
+    )
+    build_job_is_trcomp_dedicated = os.getenv("TRCOMP_DEDICATED", "false").lower() == "true"
+
+    return not (
+        build_job_is_ei_dedicated
+        or build_job_is_neuron_dedicated
+        or build_job_is_neuronx_dedicated
+        or build_job_is_graviton_dedicated
+        or build_job_is_habana_dedicated
+        or build_job_is_hf_trcomp_dedicated
+        or build_job_is_trcomp_dedicated
+    )
+
+
+def does_dev_config_enable_any_build_modes():
+    """
+    Return true if the dev config file enables any specific build mode
+    :return: bool True/False
+    """
+    dev_config_enables_ei_build_mode = parse_dlc_developer_configs("dev", "ei_mode")
+    dev_config_enables_neuron_build_mode = parse_dlc_developer_configs("dev", "neuron_mode")
+    dev_config_enables_neuronx_build_mode = parse_dlc_developer_configs("dev", "neuronx_mode")
+    dev_config_enables_graviton_build_mode = parse_dlc_developer_configs("dev", "graviton_mode")
+    dev_config_enables_habana_build_mode = parse_dlc_developer_configs("dev", "habana_mode")
+    dev_config_enables_hf_trcomp_build_mode = parse_dlc_developer_configs("dev", "hf_trcomp_mode")
+    dev_config_enables_trcomp_build_mode = parse_dlc_developer_configs("dev", "trcomp_mode")
+
+    return (
+        dev_config_enables_ei_build_mode
+        or dev_config_enables_neuron_build_mode
+        or dev_config_enables_neuronx_build_mode
+        or dev_config_enables_graviton_build_mode
+        or dev_config_enables_habana_build_mode
+        or dev_config_enables_hf_trcomp_build_mode
+        or dev_config_enables_trcomp_build_mode
+    )
+
+
+def is_training_or_inference_enabled_for_this_pr_build():
```

Comment:
Ack. Docstrings are missing in general because I'm trying to fix bugs that occur during execution, and the state of the code is in flux.

Will add docstrings after getting the code working fully.